[Rishabh Anand] iP

rish-16 opened the pull request:
## Updated PR Description for Week 4

> *"I am Iron Man"* ~ Tony Stark, Avengers Endgame, 2019

### Breakfast
Here's a list of stuff I had for breakfast today:
- Peanut Butter sandwich
- Mango juice
- Cinnamon Bun
- Chocolate Rolls

### Favourite Bands
Here's my ranking of my top 3 *Heavy Metal* bands:
1. Iron Maiden
2. Avenged Sevenfold
3. Black Sabbath

### Did you know?
Here's a fun snippet of code:

```python
class Module(object):
    def __init__(self, mod_code, prof_name, semester_offered):
        self.mod_code = mod_code
        self.prof_name = prof_name
        self.semester_offered = semester_offered

    def __repr__(self):
        return "{} taught by {} in {}".format(self.mod_code, self.prof_name, self.semester_offered)
```

Front page of the Internet? You're looking for **[Reddit](http://reddit.com)**!

### To-do list
- [ ] Do CS2103 `iP`
- [ ] Do CS2103 `tP`
- [x] Call Mom

Commit:
Add Gradle support

rish-16 committed:
Add Level-1 chatbot

rish-16 committed:
Add Level-2 to-do list

rish-16 committed:
Add option status check

rish-16 committed:
Add Task class for abstraction

rish-16 committed:
Add multiple types of tasks

rish-16 committed:
Add module tests

rish-16 committed:
Add module testing

rish-16 committed:
Add and handle exceptions

rish-16 committed:
Add task deletion feature

rish-16 committed:
Separate exceptions into function

rish-16 committed:
Add comments to methods

rish-16 committed:
Add comments for class

rish-16 committed:
Add advanced syntax exception handling

rish-16 committed:
Add new test cases

rish-16 committed:
Add minor updates to tests

rish-16 committed:
Merge Level 7 and 8 branches and respective features

rish-16 committed:
Merge two branches for Level 7 and 8

rish-16 committed:
Add duke.txt master file

rish-16 committed:
Fix duke.txt merge conflict between branches

rish-16 committed:
Refactor Duke class and make chatbot more modular

rish-16 committed:
Add package information and bundle code together

rish-16 committed:
Add package information and bundle code together

rish-16 committed:
Add JUnit tests for duke package

rish-16 committed:
Complete JAR-compatible package

rish-16 committed:
Add code comments and fix formatting

rish-16 committed:
Add find text feature for Level 9

rish-16 committed:
Add duke.txt flat file

rish-16 committed:
Add updates to README

rish-16 committed:
Add updates to README

rish-16 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

lhw-1 reviewed (commented):
LGTM. No coding standard violations found. Excellent work!

rish-16 committed:
Update gradle to enable automated builds

jonas-chow commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+        }
+    }
+
+    public void run() {
```
Perhaps a javadoc comment should be added here (and for all other public methods)?

jonas-chow commented on src/main/java/duke/parser/Brain.java:
```diff
@@ -0,0 +1,203 @@
+            Task task = dataStore.get(i);
+
+            if (task.getStatus()) {
+                userInt.say(i+1 + ". " + task.toString());
```
There should be spaces surrounding operators for better readability

jonas-chow commented on src/main/java/duke/parser/Brain.java:
```diff
@@ -0,0 +1,203 @@
+    public void shutdownBot(Ui userInt, DataStore dataStore, MemoryBuffer memBuff) {
+        // save current task log 
+        try {
+            memBuff.writeFile(dataStore);
```
Should the writing to file only be executed on shutdown bot? What if the program gets interrupted through other means (e.g. accidentally closing the window, running out of power, ctrl + c)? Consider saving after every operation instead.

jonas-chow commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+package duke.tasks;
```
The order of your package statement and the class-level javadoc comment is inconsistent here. Given that most of your files have the comment before the package, consider shifting this package statement downwards?

jonas-chow commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,36 @@
+                + super.toString()
+                + " (at: "
+                + this.timing
+                + ")";
```
Consider condensing this into 2 lines at most. Slightly excessive line breaking.

jonas-chow commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,61 @@
+ */
+abstract public class Task {
+    protected String description;
+    protected Boolean status;
```
Consider using the primitive type `boolean` instead for performance and ease of coding

jonas-chow commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,61 @@
+            return true;
+        } else {
+            return false;
+        }
```
Can be reduced into a simple `return t.getDescription().equals(this.description);`

jonas-chow commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,51 @@
+     * Prints out the welcome greeting upon successful summoning.
+     */
+    public void greet() {
+        System.out.println("Hello, I\'m  JARVIS");
```
Backslash to escape not required here since they use different quotation marks.

jonas-chow reviewed (commented):
LGTM, mostly. A few minor coding standard violations that can be easily fixed. 

Some files like Brain are have relatively many lines of code, and are vaguely named. 
In my opinion, it might be better to split it up into several smaller components with more concrete names that are easier to understand so that it's easier to find and edit particular spots in the future. 
For example, perhaps you could have one class for each command, that handles the brain's functionality, as well as the parser's error handling functionality.

rish-16 committed:
Create GUI for JarVIS chatbot with full functionality

rish-16 commented:
Thanks @jonas-chow was considering creating different files for each function of the Brain

rish-16 committed:
Add JAR file for JarVIS bot

This commit allows users to download and run the most recent
version of the JarVIS task management bot, powered by an intuitive
text-based user interface.

rish-16 committed:
Add assertions to Brain processor

This commit ensures that JarVIS's Brain actually performs the tasks
in the right manner without updating the wrong variables or data
containers.

rish-16 committed:
Refactor code to improve code quality

This commit refactors code that added a lot of overhead to the project. JarVIS's codebase has been cleaned and warnings have been removed to ensure a smooth run.

rish-16 committed:
Fix UI dimensions and add user manual command

rish-16 committed:
Fix alignment issues of dialog box

rish-16 committed:
Upgrade UI to make chatbot more aesthetic

rish-16 committed:
Update user guide with features and instructions; Add UI screenshot to docs

rish-16 committed:
Update image on README

rish-16 committed:
Add file creation ability if file doesn't exist

rish-16 committed:
Update Manifest after build

rish-16 committed:
Remove redundant file checks

rish-16 committed:
Add final working version of JarVIS

rish-16 committed:
Fix datetime parsing bugs; Remove inability to send commands

rish-16 committed:
Update UG with feature guide
